include/cuco/detail/static_set/static_set_ref.inl

```diff
@@ -210,6 +210,45 @@ auto static_set_ref<Key, Scope, KeyEqual, ProbingScheme, StorageRef, Operators..
     std::move(*this)};
 }
 
+template <typename Key,
+          cuda::thread_scope Scope,
+          typename KeyEqual,
+          typename ProbingScheme,
+          typename StorageRef,
+          typename... Operators>
+template <typename NewKeyEqual>
+__host__ __device__ constexpr auto
+static_set_ref<Key, Scope, KeyEqual, ProbingScheme, StorageRef, Operators...>::with_key_eq(
+  NewKeyEqual const& key_equal) const noexcept
+{
+  return static_set_ref<Key, Scope, NewKeyEqual, ProbingScheme, StorageRef, Operators...>{
+    cuco::empty_key<Key>{this->empty_key_sentinel()},
+    key_equal,
+    this->impl_.probing_scheme(),
+    {},
+    this->impl_.storage_ref()};
+}
+
+template <typename Key,
+          cuda::thread_scope Scope,
+          typename KeyEqual,
+          typename ProbingScheme,
+          typename StorageRef,
+          typename... Operators>
+template <typename NewHash>
+__host__ __device__ constexpr auto
+static_set_ref<Key, Scope, KeyEqual, ProbingScheme, StorageRef, Operators...>::with_hash_function(
+  NewHash const& hash) const noexcept
+{
+  auto const probing_scheme = this->impl_.probing_scheme().with_hash_function(hash);
```

Comment:
This surfaces the old problem with `double_hashing` taking two hash functions while the other probing schemes take a single hash function. Would it make sense to make the `NewHash` tparam variadic, i.e., `NewHashFunctions…`?